server/webhook.go

```diff
@@ -158,6 +181,10 @@ func (p *Plugin) postPullRequestEvent(event *github.PullRequestEvent) {
 			continue
 		}
 
+		if isPrivate && !p.permissionToRepo(sub.CreatorID, repoName) {
```

Comment:
Is it possible to centeralize these checks? They all do similar things but are all spread out. Is it possible to just do the check in `handleWebhook`?

Reply:
Possibly with a significant amount of refactoring

Reply:
@crspeller pushed some refactoring that should clean it up a bit